appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/BaseMutableUiState.kt

```diff
@@ -24,13 +29,14 @@ import kotlinx.coroutines.flow.update
 abstract class BaseMutableUiState<TargetUiState>(
     val uiContext: UiContext,
     val motionProperties: List<MotionProperty<*, *>>
-) {
+) : TransitionBoundsAware {
 
-    private val containerRect = Rect(
+    private var containerRect = Rect(
         offset = Offset.Zero,
         size = Size(
-            width = uiContext.transitionBounds.widthPx.toFloat(),
-            height = uiContext.transitionBounds.heightPx.toFloat()
+            // permissive for visibility calculation to return true on first composition
+            width = Float.MAX_VALUE,
```

Comment:
not sure about this, it will lead to lifecycle jumps to `STARTED` and then immediately after to `STOPPED`